apps/src/netsim/NetSimLocalClientNode.js

```diff
@@ -457,7 +457,13 @@ NetSimLocalClientNode.prototype.sendMessages = function (payloads, onComplete) {
  * @private
  */
 NetSimLocalClientNode.prototype.onNodeTableChange_ = function (nodeRows) {
-  // 1. Remove simulating routers that have vanished from remote storage.
+  // If our own row is gone, drop everything and handle disconnect.
+  if (this.onNodeLostConnection_ && !this.canFindOwnRowIn(nodeRows)) {
+    this.onNodeLostConnection_();
+    return;
+  }
```

Comment:
If we can't find our own row in nodeRows but we don't have an onNodeLostConnection_ handler, do we actually want to keep going?